modules/dists/BlockDist.chpl

```diff
@@ -955,6 +956,31 @@ iter BlockImpl.activeTargetLocales(const space : domain = boundingBox) {
   }
 }
 
+iter BlockImpl.activeTargetLocales(const space : range(?)) {
+  compilerAssert(rank==1);
+  const dims = targetLocsIdx(space.first)..targetLocsIdx(space.last);
+
+  // In case 'space' is a strided domain we need to check that the locales
+  // in 'dims' actually contain indices in 'locSpace'.
+  //
+  // Note that we cannot use a simple stride here because it is not guaranteed
+  // that each locale contains the same number of indices. For example, the
+  // domain {1..10} over four locales will split like:
+  //   L0: -max(int)..3
+  //   L1: 4..5
+  //   L2: 6..8
+  //   L3: 9..max(int)
+  //
+  // The subset {1..10 by 4} will involve locales 0, 1, and 3.
+  foreach i in dims {
+    const chunk = chpl__computeBlock(i, targetLocDom, boundingBox, boundingBox.dims());
+    // TODO: Want 'contains' for a domain. Slicing is a workaround.
+    if space[chunk[0]].sizeAs(int) > 0 then
```

Comment:
looks like we have it: https://chapel-lang.org/docs/language/spec/domains.html#ChapelDomain.contains

Reply:
You're right. I'll give that a go. This TODO is something I inherited from the current version not mine. (But I can still test it while here)

Reply:
I think what this comment meant is "overlaps". The check here is not whether `chunk` is wholly contained in `space`, which is what `contains` does. All we check for is any overlap, for which we don't have a helper method as far as I know.

Reply:
I adjusted the comment.